gemini/sample-apps/fashion-trends-prediction/app/articles.py

```diff
@@ -0,0 +1,122 @@
+import json
+import pickle
+
+import vertexai
+from config import config
+from langchain.docstore.document import Document
+from langchain.retrievers import BM25Retriever, EnsembleRetriever
+from langchain.vectorstores import VectorStore
+from sentence_transformers import SentenceTransformer
+from vertexai.language_models import ChatModel, InputOutputTextPair
```

Comment:
It's preferred to use Gemini instead of PaLM whenever possible.

Reply:
Updated - Gemini is used everywhere now.